--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -1278,6 +1278,12 @@
       chrome_dylib_version,
     ]
 
+    # This non-component development build contains four unwind personalities.
+    # Apple's compact format has only three slots; retain DWARF unwind instead.
+    if (!use_lld && !is_component_build && !is_official_build) {
+      ldflags += [ "-Wl,-no_compact_unwind", "-Wl,-keep_dwarf_unwind" ]
+    }
+
     if (!is_component_build) {
       # Specify a sensible install_name for static builds. The library is
       # dlopen()ed so this is not used to resolve the module.
